tests/scalers/temporal/temporal_test.go

```diff
@@ -0,0 +1,238 @@
+  labels:
+    app: {{.DeploymentName}}
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: {{.DeploymentName}}
+  template:
+    metadata:
+      labels:
+        app: {{.DeploymentName}}
+    spec:
+      containers:
+      - name: worker
+        image: "temporaliotest/omes:go-ci-latest"
+        imagePullPolicy: Always
+        command: ["/app/temporal-omes"]
+        args:
+        - "run-worker"
+        - "--language=go"
+        - "--server-address={{.TemporalDeploymentName}}.{{.TestNamespace}}.svc.cluster.local:7233"
+        - "--run-id=test"
+        - "--scenario=workflow_with_single_noop_activity"
+        - "--dir-name=prepared"
+`
+
+	jobWorkFlowTemplate = `
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: workflow
+  namespace: {{.TestNamespace}}
+spec:
+  template:
+    spec:
+      containers:
+      - name: workflow
+        image: "temporaliotest/omes:go-ci-latest"
+        imagePullPolicy: Always
+        command: ["/app/temporal-omes"]
+        args:
+        - "{{.WorkFlowCommand}}"
+        {{- if eq .WorkFlowCommand "run-scenario"}}
+        - "--iterations={{.WorkFlowIterations}}"
+        {{- end}}
+        - "--scenario=workflow_with_single_noop_activity"
+        - "--run-id=test"
+        - "--server-address={{.TemporalDeploymentName}}.{{.TestNamespace}}.svc.cluster.local:7233"
+      restartPolicy: OnFailure
+  backoffLimit: 10
+`
+)
+
+func getTemplateData() (templateData, []Template) {
+	return templateData{
+			WorkFlowCommand:        "run-scenario",
+			WorkFlowIterations:     2,
+			TestNamespace:          testNamespace,
+			TemporalDeploymentName: TemporalDeploymentName,
+			ScaledObjectName:       scaledObjectName,
+			DeploymentName:         deploymentName,
+		}, []Template{
+			{Name: "scaledObjectTemplate", Config: scaledObjectTemplate},
+			{Name: "deploymentTemplate", Config: deploymentTemplate},
+		}
+}
+
+func TestTemporalScaler(t *testing.T) {
+	kc := GetKubernetesClient(t)
+	data, templates := getTemplateData()
+	CreateNamespace(t, kc, testNamespace)
+
+	KubectlApplyWithTemplate(t, data, "temporalServiceTemplate", temporalServiceTemplate)
+	KubectlApplyWithTemplate(t, data, "temporalDeploymentTemplate", temporalDeploymentTemplate)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, TemporalDeploymentName, testNamespace, 1, 30, 4), "temporal is not in a ready state")
+
+	KubectlApplyMultipleWithTemplate(t, data, templates)
+	testActivation(t, kc, data)
+	testScaleOut(t, kc, data)
+	testScaleIn(t, kc, data)
+	DeleteKubernetesResources(t, testNamespace, data, templates)
+}
+
+func testActivation(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing activation ---")
+
+	KubectlApplyWithTemplate(t, data, "jobWorkFlowActivation", jobWorkFlowTemplate)
+	assert.True(t, WaitForJobCount(t, kc, testNamespace, 1, 60, 3), "job count in namespace should be 1")
+
+	AssertReplicaCountNotChangeDuringTimePeriod(t, kc, deploymentName, testNamespace, 0, 180)
```

Comment:
1 minute is enough :) 
```suggestion
	AssertReplicaCountNotChangeDuringTimePeriod(t, kc, deploymentName, testNamespace, 0, 60)
```